crypto/types/src/types.rs

```diff
@@ -817,6 +834,53 @@ impl PublicKey {
             compressed: new_pk.to_encoded_point(true),
         }
     }
+
+    /// Generates a new random public key.
+    /// Because the corresponding private key is discarded, this might be useful only for testing purposes.
+    pub fn random() -> Self {
+        let (_, pk) = Self::random_keypair();
```

Comment:
Btw pk could be private key or public key.

Reply:
... the code is unchanged. Just merged two `impl` sections. but good point

Reply:
Yeah, but let's fix it, it should at least be called `public_key`.

Reply:
If this is also for testing, consider introducing the `#[cfg(test)]` for the function.

Reply:
good point, also added to `HalfKey::random()`